texastribune/queso-ui
# Add better documentation endpoints

#### What's this PR do?

Converts the style-doc logic to use typescript which helps breaks down various parts of the style system (e.g. colors, modifiers, tokens) into different [types](https://github.com/texastribune/queso-ui/blob/new-endpoint-2/docs/config/tasks/types.ts).

##### Classes added (if any)
None
##### Classes removed (if any)
None


#### Why are we doing this? How does it help us?

TBH, this is probably complete overkill. I've been wanting to clean up that SCSS [comment parsing step](https://github.com/texastribune/queso-ui/blob/new-endpoint-2/docs/config/tasks/style-doc.ts), which is what builds out the CSS data for the docs and this just kind of helped me bring order to that chaos. 

@AndrewGibson27, there's a ton of code here and I don't want you to spend too much time sorting through it. The `style-doc.ts` part is the only real change and it only pertains to the docs — nothing that affects where queso styles are used. I'm sure I broke all the typescript rules along the way. 

FWIW `npm run dev` is faster with the cleaner, more structured style data.

#### How should this be manually tested?
`npm install`
`npm run dev`

Just confirm the docs are loading properly and you don't run into any build errors. 
There's a little blurb about the endpoints at the bottom of the about page, which should help sort through the new data output: http://localhost:8080/about/


#### Does this introduce a breaking change where queso-ui is used in the wild? If so, is there a relevant branch/PR to accompany this release?

Doesn't even merit a release
